Sources/BraintreeSEPADirectDebit/BTSEPADirectDebitClient.swift

```diff
@@ -36,16 +43,29 @@ import BraintreeCore
             if error != nil {
                 completion(nil, error)
                 return
-            } else if result != nil {
-                // TODO: future PR start ASWebAuthenticationSession with result.approvalURL
+            } else if result != nil, let result = result {
+                if result.approvalURL == "null" {
+                    // TODO: call tokenize - url already approved
+                } else if let url = URL(string: result.approvalURL) {
+                    self.startAuthenticationSession(url: url, context: context) { success, error in
+                        switch success {
+                        case true:
+                            // TODO: call BTSEPADirectDebitClient.tokenize
+                            return
+                        case false:
+                            completion(nil, error)
+                            return
+                        }
+                    }
+                }
             }
```

Comment:
We should add another else here. If `error` is nil and `result` is nil, we should pass a generic error in the completion.